framework/dependency/dependency.go

```diff
@@ -0,0 +1,32 @@
+package dependency
+
+import (
+	"sync"
+)
+
+// Wrapper around WaitGroups to facilitate waiting on dependencies to be ready
+type dependency struct {
+	// WaitGroup used internally to store and manage the ready state
+	waiter sync.WaitGroup
+}
+
+// Create new instance of dependency, initially set to not-ready state
+func New() *dependency {
+	var dep dependency
+	// Initially increment WaitGroup so that all calls to Await() will lock
+	dep.waiter.Add(1)
+	return &dep
+}
```

Comment:
```suggestion
func New() dependency {
	var dep dependency
	// Initially increment WaitGroup so that all calls to Await() will lock
	dep.waiter.Add(1)
	return dep
}
```

You should return values, not pointers.

Reply:
Fixed